src/traces/pie/plot.js

```diff
@@ -731,7 +734,7 @@ function calcMaxHalfSize(a, halfAngle, r, ring) {
 }
 
 function getInscribedRadiusFraction(pt, cd0) {
-    if(pt.v === cd0.vTotal && !cd0.trace.hole) return 1;// special case of 100% with no hole
+    if(pt.v === cd0.vTotal && !(cd0.trace.hole[0] || cd0.trace.hole)) return 1;// special case of 100% with no hole
```

Comment:
Using `Lib.isArrayOrTypedArray` let's test `trace.hole` to be an array before try accessing index 0.

Reply:
It's not necessarily the case that a full-circle slice is the first slice, there could be a zero-size slice before it. So (once we know `hole` is an array) rather than `[0]` we should use the index of this slice, which I believe is `[pt.i]`.

Reply:
I've made the change, please double check that this makes sense :) I'm not 100% sure how to handle this edge case.